### PR TITLE
flow improvements

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform gcp

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/controller/infrastructure/flow_reconciler.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+		Client:         f.client,
+		PersistFunc: func(ctx context.Context, state *runtime.RawExtension) error {
+			return patchProviderStatusAndState(ctx, f.client, infra, nil, state)
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create flow context: %v", err)
+	}
+
+	status, state, err := fctx.Reconcile(ctx)
```
Comment (hebelsan) [inline pkg/controller/infrastructure/flow_reconciler.go:74]: Can we maybe come up with something more meaningful then status and state.
Some suggestions: privState, currentState, progState, progress...

File: pkg/controller/infrastructure/actuator.go [hunk 2]
```diff
@@ -6,22 +6,17 @@ package infrastructure
```
[the remark this answers is not in the document]
Comment (kon-angelo) [reply pkg/controller/infrastructure/actuator.go:6]: the `actuator.go` is the entry point for both reconcilers, so I do not necessarily agree here.

File: pkg/controller/infrastructure/infraflow/ensure.go [hunk 3]
```diff
@@ -497,28 +461,30 @@ func (c *FlowContext) ensureFirewallRulesDeleted(ctx context.Context) error {
...
+func (fctx *FlowContext) ensureKubernetesRoutesDeleted(ctx context.Context) error {
 	log := shared.LogFromContext(ctx)
-	vpcName := c.vpcNameFromConfig()
-
-	var names []string
-	routes, err := c.computeClient.ListRoutes(ctx)
+	vpcName := fctx.vpcNameFromConfig()
+
+	routes, err := fctx.computeClient.ListRoutes(ctx, client.RouteListOpts{
+		// Filter: fmt.Sprintf("network~'%s'", vpcName),
```
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/ensure.go:469]: Leftover from development? If so rm. If this is here for future reference consider adding a few words as to why

File: pkg/controller/infrastructure/infraflow/ensure.go [hunk 4]
```diff
@@ -497,28 +461,30 @@ func (c *FlowContext) ensureFirewallRulesDeleted(ctx context.Context) error {
...
-			if strings.HasPrefix(urlParts[len(urlParts)-1], c.clusterName) {
-				names = append(names, route.Name)
-			}
-		}
-	}
-
-	for _, name := range names {
-		log.Info(fmt.Sprintf("destroying route[name=%s]", name))
-		err := c.computeClient.DeleteRoute(ctx, name)
+		log.Info(fmt.Sprintf("destroying route[name=%s]", route.Name))
```
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/ensure.go:486]: missing whitespace in `route[name=…`

File: pkg/controller/infrastructure/infraflow/reconciler.go [hunk 5]
```diff
@@ -164,67 +170,67 @@ func (c *FlowContext) getStatus() *v1alpha1.InfrastructureStatus {
...
-			CreatedResourcesExistKey: ptr.Deref(c.whiteboard.Get(CreatedResourcesExistKey), ""),
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+			Kind:       "InfrastructureState",
 		},
+		Data: fctx.whiteboard.ExportAsFlatMap(),
+		// Data: map[string]string{
+		//
+		// 	CreatedResourcesExistKey: ptr.Deref(fctx.whiteboard.Get(CreatedResourcesExistKey), ""),
+		// },
```
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/reconciler.go:211]: Rm, unless you plan to use it in future

File: pkg/controller/infrastructure/infraflow/shared/whiteboard.go [hunk 6]
```diff
@@ -271,6 +271,7 @@ func IsValidValue(value string) bool {
 	return value != "" && value != deleted
 }
 
+// ObjectKeys returns a slice containing the keys of the Whiteboard in ascending order.
```
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/shared/whiteboard.go:274]: That should be in l.287 I think

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Pull request merged**